creed/rosetta.py

```diff
@@ -0,0 +1,290 @@
+                os.makedirs(raw_folder)
+
+            for k in range(batch_data.shape[-1]):
+                channel_name = batch_data.channels.values[k] + '.tiff'
+
+                # save tifs to appropriate directories
+                if save_format in ['normalized', 'both']:
+                    save_path = os.path.join(norm_folder, channel_name)
+                    io.imsave(save_path, comp_data[j, :, :, k] / norm_const, check_contrast=False)
+
+                if save_format in ['raw', 'both']:
+                    save_path = os.path.join(raw_folder, channel_name)
+                    io.imsave(save_path, comp_data[j, :, :, k], check_contrast=False)
+
+# TODO: Decide if this code is necessary or not
+# def compare_comped_images(raw_dir, comp_dir_list, output_dir):
+#     """Creates a tiled image containing the raw image, difference image, and output image
+#     from each compensated directory supplied
+#
+#     Args:
+#         raw_dir: directory containing raw images
+#         comp_dir_list: list of directories containing compensated images
+#         output_dir: directory where tifs will be saved"""
+#
+#     # load images
+#     raw_images = load_imgs_from_tree(raw_dir, dtype='float')
+#     comp_dict = {}
+#     for dir_name in comp_dir_list:
+#         comp_images = load_imgs_from_tree(dir_name, dtype='float')
+#         comp_dict[dir_name] = comp_images
+#
+#     img_size = raw_images.shape[1]
+#     fov_num = raw_images.shape[0]
+#     # compute difference between first compensation image and baseline
+#     diff_images = raw_images.values - comp_dict[comp_dir_list[0]].values
+#     diff_images[diff_images < 0] = 0
+#
+#     comp_image_num = len(comp_dir_list)
+#     # loop over each channel
+#     for j in range(raw_images.shape[3]):
+#         # create tiled array of corrections x fovs
+#         tiled_image = np.zeros((img_size * (comp_image_num + 2),
+#                                 img_size * fov_num))
+#         # loop over each fov, and place into columns of tiled array
+#         for i in range(raw_images.shape[0]):
+#             start = i * img_size
+#             end = (i + 1) * img_size
+#
+#             # first row is raw image
+#             tiled_image[:img_size, start:end] = raw_images.values[i, :, :, j]
+#
+#             # second row is difference image between raw and first compensated image
+#             tiled_image[img_size:(img_size * 2), start:end] = diff_images[i, :, :, j]
+#
+#             # subsequent rows are compensated images with different coefficients
+#             for idx, key in enumerate(comp_dir_list):
+#                 tiled_image[(img_size * (idx + 2)):(img_size * (idx + 3)), start:end] = \
+#                     comp_dict[key].values[i, :, :, j]
+#
+#         io.imsave(os.path.join(output_dir, raw_images.channels.values[j] + '_comparison.tiff'),
+#                   tiled_image)
+
+
+def create_tiled_comparison(input_dir_list, output_dir):
```

Comment:
Yeah, it's pretty flexible. Any sort of comparison across runs would benefit from this. Probably a good candidate for the toolbox repo once we create it. For now I'm going to keep it here since no one else is using it